video_xblock/mixins.py

```diff
@@ -555,22 +570,24 @@ def settings(self):
                     "account_id": "1234567890"
                 }
         """
-        s_service = self.runtime.service(self, 'settings')
-        if s_service:
-            # At the moment SettingsService is not available in the context
-            # of Studio Edit modal. See https://github.com/edx/edx-platform/pull/14648
-            return s_service.get_settings_bucket(self)
+        site_name = get_current_site_name()
+        if not site_name:
+            return {}
 
         settings = import_from('django.conf', 'settings')
-        return settings.XBLOCK_SETTINGS.get(self.block_settings_key, {})
+        return settings.XBLOCK_SETTINGS.get(site_name, {})
 
-    def populate_default_values(self, fields_dict):
+    def populate_default_values(self, submit_data):
         """
         Populate unset default values from settings file.
         """
         for key, value in self.settings.items():
-            fields_dict.setdefault(key, value)
-        return fields_dict
+            submit_data.setdefault(key, value)
+            # if key is present in submit data but set to empty value:
+            if key in submit_data and not submit_data[key]:
+                submit_data[key] = value
```

Comment:
`dict.update()` overrides existed values but wanna to set only those which are empty or are absent.